pandas/plotting/_matplotlib/hist.py

```diff
@@ -417,11 +419,21 @@ def hist_frame(
         if not isinstance(column, (list, np.ndarray, ABCIndexClass)):
             column = [column]
         data = data[column]
-    data = data._get_numeric_data()
+    # GH32590
+    columns_copy = data.columns
+    numeric_data = data._get_numeric_data()
+    datetime_data = data.select_dtypes(include=["datetime64", "datetimetz"])
```

Comment:
you can just add on integer, float (and add exclude=['timedelta'])

Reply:
I think `include=[int, float]` is not enough to replace `_get_numeric_data()`, for numeric types such as `uint32` won't be selected by `int` (but currently it works well with `pd.DataFrame.hist`).

As to `timedelta`, experimentally `timedelta` is not supported yet, but we can convert it into float to solve this problem, see

> As mentioned in #36287 (comment), converting datetimetz, datetime64 and timedelta into np.int64 (or using map/dict, just like what #32590 (comment) has discussed before) may be a simpler choice, which can leave all tests unchanged.

BTW, which test file (`test_hist_method.py`/`test_frame.py`/`test_datetimelike.py`) is better to put new tests required in https://github.com/pandas-dev/pandas/pull/36287#issuecomment-692921341 in ? @jreback @charlesdong1991

Reply:
> include=[np.number] is not qualified too

what if include `np.number` and exclude `timedelta`? will the behaviour remain unchanged and all tests can pass?

> which test file (test_hist_method.py/test_frame.py/test_datetimelike.py) is better to put new tests

should be `test_hist_method`

Reply:
> what if include `np.number` and exclude `timedelta`? will the behaviour remain unchanged and all tests can pass?

It works, I'll change to it